azavea/franklin
# Update collection extents when items change

## Overview

This PR makes it so that when you add an item via the POST endpoint or add lots of items with the bulk importer, the collection extent it grown to include any new geometric area that's covered.

It's wip because on update / delete you have to check all the existing items to calculate the new extent, and I'd like to do that in
the background rather than in a way that blocks the response, in case it's costly. 

### Checklist


### Notes

I'm scoping this down to just _expanding_ extends when items are added or changed. The reason for this is that in those two cases, I always have enough information from the new items to calculate the new extent. If I want to shrink an extent, then I need to know the spatial and temporal extent of every item in this collection in the database to figure out the new extent. I don't know how costly that aggregation will be, so I'd like to do it in the background to protect us from large updates, but at the same time I'm not certain that doing it in the background will be sufficient since the transaction will acquire a write lock against the collection row.

### Testing Instructions

This is pretty hard to test unless you have data handy already, but here's the testing flow I did, and which I'm happy to demo as well:

- save off an existing collection into a new json file
- update the id to `dummy-collection`
- post it to collections
- grab a nice item from the prior collection and save it off to a new json file
- edit its geometry and datetime to be outside the current spatial and temporal extent of the new collection you just created
- post it to `/collections/dummy-collection/items`
- fetch your collection again -- the extent should now include the time and bbox of the item you just posted 🎉

Closes #615
